torchvision/transforms/transforms.py

```diff
@@ -47,13 +51,30 @@ class Compose(object):
     """
 
     def __init__(self, transforms):
-        self.transforms = transforms
+        super().__init__()
 
-    def __call__(self, img):
+        try:
+            # Try to wrap the transforms that are not nn.Module into nn.Lambda
+            # register all transforms as nn.ModuleList <-> all transforms should be nn.Module
+            # if something fails we keep previous behaviour
+            new_transforms = []
+            for t in transforms:
+                if callable(t) and not isinstance(t, torch.nn.Module):
+                    t = Lambda(t)
+                new_transforms.append(t)
+
+            self.transforms = torch.nn.ModuleList(new_transforms)
+        except TypeError:
+            self.transforms = transforms
```

Comment:
I think you are a little too paranoid here. The only time we could hit a `TypeError` is if `t` is not a callable. It is true, that this raised no error before, but the resulting `Compose` would not have been valid: if you call it in such case L72 would raise the error during the forward pass. Thus, anyone that is affected by now raising the `TypeError` at instantiation wasn't using the `Compose` anyway.

```suggestion
        new_transforms = []
        for t in transforms:
            if not callable(t):
                raise TypeError
            if not isinstance(t, torch.nn.Module):
                t = Lambda(t)
            new_transforms.append(t)

        self.transforms = torch.nn.ModuleList(new_transforms)
```